### ISSUE TITLE
Incorrect warning RS1024 Compare symbols correctly

### ISSUE THREAD

**Issue by jods4** [dotnet/roslyn#42758]
**Version Used**: 3.5.0

**Steps to Reproduce**:
Compare two `ISymbol` through their `IEquatable<ISymbol>` interface
```csharp
ISymbol s1, s2;
// hovering indicates this is not Object.Equals(object) 
// but IEquatable<ISymbol>.Equals(ISymbol)
var eq = s1.Equals(s2); 
```

**Expected Behavior**:
No warning.

**Actual Behavior**:
Warning RS1024 _Compare symbols correctly_

**Comment by sharwell** [comment]
This is the expected behavior. When building against a version of roslyn that provides `SymbolEqualityComparer`, it should be used instead of `Equals` for comparing symbols.
